test/e2e/shared/common.go

```diff
@@ -214,51 +214,54 @@ func (o OpenStackLogCollector) CollectMachineLog(ctx context.Context, management
 		return fmt.Errorf("error writing server JSON %s: %s", serverJSON, err)
 	}
 
-	srvUser := o.E2EContext.E2EConfig.GetVariable(SSHUserMachine)
-	executeCommands(
-		ctx,
-		o.E2EContext.Settings.ArtifactFolder,
-		o.E2EContext.Settings.Debug,
-		outputPath,
-		ip,
-		openStackCluster.Status.Bastion.FloatingIP,
-		srvUser,
-		[]command{
-			// don't do this for now, it just takes to long
-			// {
-			//	title: "systemd",
-			//	cmd:   "journalctl --no-pager --output=short-precise | grep -v  'audit:\\|audit\\['",
-			// },
-			{
-				title: "kern",
-				cmd:   "journalctl --no-pager --output=short-precise -k",
+	// If the machine has a bastion, collect logs from there.
+	if openStackCluster.Status.Bastion != nil {
```

Comment:
I'm confused about this. I mean the code makes sense to me but I'm confused about if we have bastions or not. Yesterday I thought that we didn't have bastions when I saw you added the bastion patches, but now when I see this and realize that we cannot collect logs without bastions, then we must have bastions currently because we definitely do have these logs!

What I see now in the tests run on this PR is that there are no longs for the machines (e.g. [here](https://gcsweb.k8s.io/gcs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-openstack/1796/pull-cluster-api-provider-openstack-e2e-test/1745523472215314432/artifacts/clusters/bootstrap/e2e-4p833k/machines/)). But in our periodic tests (e.g. [this](https://prow.k8s.io/view/gs/kubernetes-jenkins/logs/periodic-cluster-api-provider-openstack-e2e-test-main/1745503729274064896)) there [are logs](https://gcsweb.k8s.io/gcs/kubernetes-jenkins/logs/periodic-cluster-api-provider-openstack-e2e-test-main/1745503729274064896/artifacts/clusters/bootstrap/e2e-6urfd4/machines/) for some machines at least.
So now it seems to me that we do have bastions in at least some tests and this PR is perhaps removing them and consequently prevents log collection.

Reply:
I feel like I'm on the same rollercoaster. Weeeeeeeeeee.